src/pudl/output/eia923.py

```diff
@@ -126,6 +126,30 @@ def _fill_fuel_costs_by_state(
     return out_df
 
 
+def drop_ytd_for_annual_tables(df: pd.DataFrame, freq: str) -> pd.DataFrame:
+    """Drop records in annual tables where data_maturity is incremental_ytd.
+
+    This avoids accidental aggregation errors due to sub-annually reported data.
+
+    Args:
+        df: A pd.DataFrame that contains a data_maturity column and for
+            which you want to drop values where data_maturity = incremental_ytd.
+        freq: either MS or AS to indicate the level of aggretation for a specific table.
+
+    Returns:
+        pd.DataFrame: The same input pd.DataFrames but without any rows where
+            data_maturity = incremental_ytd.
+    """
+    if freq == "AS":
+        logger.info(
+            "Removing rows where data_maturity is incremental_ytd to avoid "
+            "aggregation errors."
+        )
+        return df.loc[df["data_maturity"] != "incremental_ytd"]
+    else:
+        return df
```

Comment:
this is a total nit but:
```suggestion
        df = df.loc[df["data_maturity"] != "incremental_ytd"]
    return df
```

irrc its best practice to attempt to keep the return only in one place.

and maybe it needs a  `.copy()`?